pkg/skaffold/deploy/helm_test.go

```diff
@@ -301,11 +242,10 @@ func TestHelmDeploy(t *testing.T) {
 			builds:      testBuilds,
 		},
 		{
-			description: "deploy error unmatched parameter",
+			description: "deploy should not error for unmatched parameter",
```

Comment:
This behavioral change seems unrelated to labels. Why was that necessary?

Reply:
i added the context in the description #5 of the PR
```
5. During helm deploy, build is assumed and hence if no build-artifacts
are supplied, it fails with following error
"no build present for gcr.io/k8s-skaffold/skaffold-helm"
Since Build and Deploy are now separate ( #922 for notes) use the
image value as is if not present as build artifacts.
```